src/components/settings/SettingsSSOTab/NewExistingSSOConfigs.jsx

```diff
@@ -137,6 +149,20 @@ const NewExistingSSOConfigs = ({
         config => (config.submitted_at && !config.configured_at) || (config.configured_at < config.submitted_at),
       );
       const untested = res.data.filter(config => !config.validated_at || config.validated_at < config.configured_at);
+      const timedOut = res.data.filter(
+        config => config.submitted_at && (!config.configured_at || config.submitted_at > config.configured_at) && (
+          !config.is_pending_configuration) && (!!config.errored || config.submitted_at > config.errored),
+      );
+      const errored = res.data.filter(
+        config => config.submitted_at && config.errored_at && (config.submitted_at < config.errored_at),
+      );
```

Comment:
These filter clauses are weighty enough I think they warrant helper methods at this point.

Reply:
did some organizing!